src/Hosting/Hosting/src/Startup/IRejectStartup.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    /// <summary>
+    /// A marker interface used to indicate the <see cref="IHostBuilder"/> or <see cref="IWebHostBuilder"/>
+    /// does not support any of the following extension methods:
+    /// * <see cref="WebHostBuilderExtensions.Configure(IWebHostBuilder, Action{AspNetCore.Builder.IApplicationBuilder})"/>
+    /// * <see cref="WebHostBuilderExtensions.UseStartup(IWebHostBuilder, Type)"/>
+    /// * <see cref="GenericHostWebHostBuilderExtensions.ConfigureWebHost(IHostBuilder, Action{IWebHostBuilder})"/>
+    /// </summary>
+    public interface IRejectStartup
```

Comment:
This is a new approach I'm taking instead of making `ISupportsStartup` public since it's a smaller change and it also works for `WebApplicationBuilder.Host.ConfigureWebHost`.

Reply:
This name seems weird when since this is being used to detect the host builder as well

Reply:
I'm open to suggestions.

Reply:
It's weird because interfaces themselves are about added capabilities so usually marker interfaces mark the presence of capability not the negation of it. 

- IDisallowWebHostConfiguration
- IBlockWebHostConfiguration

Reply:
`ISupportConfigureWebHost`